geomesa-native-api/src/main/java/org/locationtech/geomesa/api/GeoMesaIndex.java

```diff
@@ -0,0 +1,35 @@
+package org.locationtech.geomesa.api;
+
+import com.vividsolutions.jts.geom.Geometry;
+
+import java.util.Date;
+
+/**
+ * GeoMesaIndex is an API for utilizing GeoMesa as a spatial index
+ * without bringing in the Geotools SimpleFeature data model
+ * @param <T>
+ */
+public interface GeoMesaIndex<T> {
+
+    /**
+     * Query a GeoMesa index
+     * @param query
+     * @return
+     */
+    Iterable<T> query(GeoMesaQuery query);
+
+    /**
+     * Put a value in the GeoMesa index
+     * @param t
+     * @param geometry
+     * @param dtg
+     */
+    void put(T t, Geometry geometry, Date dtg);
```

Comment:
One idea to help with deletes below is to make the signature something like

```byte[] insert(T t, Geometry geometry, Date dtg)``` where the return ```byte[]``` or of course ```base64 String``` could be a returned ID which is compound from the Z3 bytes followed by a feature ID that you enforce the user provides. This also provides a distinct lookup for a record for delete or update.